### ISSUE TITLE
Add GeometricMean

### ISSUE THREAD

**Issue by marload** [tensorflow/addons#2019]
**Describe the feature and the current behavior/state.**

**Relevant information**
- Are you willing to contribute it (yes/no): YES
- Are you willing to maintain it going forward? (yes/no): YES
- Is there a relevant academic paper? (if so, where): NO
- Is there already an implementation in another framework? (if so, where): Pytorch Ignite Metrics
- Was it part of tf.contrib? (if so, where): NO

**Which API type would this fall under (layer, metric, optimizer, etc.)**
Metric

**Who will benefit with this feature?**
Researchers or Engineers who need a GeometricMean

**Any other info.**
None

**Comment by seanpmorgan** [comment]
Hi @marload! This seems like a good addition and should be easy to support. Please submit a PR when you have time.

**Comment by marload** [comment]
Thank You!
